document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/service/impl/AccountsServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package uk.gov.companieshouse.document.generator.accounts.service.impl;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.model.accounts.Accounts;
+import uk.gov.companieshouse.document.generator.accounts.data.accounts.AccountsManager;
+import uk.gov.companieshouse.document.generator.accounts.service.AccountsService;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+@Service
+public class AccountsServiceImpl implements AccountsService {
+
+    private static final Logger LOG = LoggerFactory.getLogger("document-generator-accounts");
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Accounts getAccounts(String resource) {
+        LOG.info("Getting accounts data from abridged-accounts-api");
+
+        ResponseEntity<Accounts> accounts = AccountsManager.getAccounts(resource);
+
+        if (accounts.getStatusCode() != HttpStatus.OK) {
+            LOG.error("Failed to retrieve data from API: " + resource);
+            return null;
```

Comment:
Is this an exceptional case? If so it may be more appropriate to throw an exception here and catch it higher up? This would save excessive null checking.

Reply:
Yeah, exception handling, logging and the abridged implementation will be done next. I'm just putting the skeleton functionality in initially and will then apply the correct exception handling and logging across the board. Currently I am putting in null checks just to keep the files down to a minimum.

Reply:
👍